h2/src/main/org/h2/mode/FunctionsOracle.java

```diff
@@ -128,10 +142,40 @@ protected Value getValueWithArgs(SessionLocal session, Expression[] args) {
         case TO_TIMESTAMP_TZ:
             result = ToDateParser.toTimestampTz(session, v0.getString(), v1 == null ? null : v1.getString());
             break;
+        case REGEXP_REPLACE:
+            Boolean replaceBackslashRefs = session.getMode().regexpReplaceBackslashReferences;
+            String regexpResult = regexpReplace(replaceBackslashRefs, v0, v1, v2, v3, v4, v5);
+            result = ValueVarchar.get(regexpResult, session);
+            break;
         default:
             throw DbException.throwInternalError("type=" + info.type);
         }
         return result;
     }
 
+    private static String regexpReplace(Boolean replaceBackslashRefs, Value ... args) {
+        String input = args[0].getString();
+        String regexp = args[1].getValueType() != Value.NULL ? args[1].getString() : "";
```

Comment:
You can simply compare result of `getString()` with `null`, there is no need to check value type.